### PR TITLE
[SQL] Bump sdk version

### PR DESCRIPTION
**Pull request by strehan1993**
**Description**

Bump SQL Version from V4 to V5 track. Implement changes needed to support new parameters.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
bump sql

**Comment by strehan1993** [comment]
It's a draft PR, please don't review it yet.

**Review by ayeletshpigelman — APPROVED** [review]

**Comment by strehan1993** [comment]
Making this PR public so that other SQL teams can commit.

File: src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py
Comment (strehan1993) [reply src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py:2733]: Please feel free to re-record and create a PR.

Comment (ericshape) [reply src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py:2733]: yea. it seems we will enable Gen4 SLO temporarily via a subscription override and complete the Gen4 tests.

**Comment by ericshape** [comment]
LGTM for firewall rules and elatic pool

**Review by ericshape — APPROVED** [review]
LGTM~
perfect work! thanks all!

**Comment by ericshape** [comment]
@Juliehzl  can you help to take a look? this PR passed all checks. thanks~

**Comment by strehan1993** [comment]
Wait, I have to push more changes.

**Comment by strehan1993** [comment]
@ericshape PR is ready now. All checks are passing. @Juliehzl please review, approve and merge with priority. This is needed for enterprise promise deadlines and is blocking some teams. Thanks!

**Review by jsntcy — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/sql/_format.py
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/sql/_format.py:255]: will these values change in the future?

File: src/azure-cli/azure/cli/command_modules/sql/_help.py
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/sql/_help.py:757]: Avoid to use real admin-sid and subscription id. Better to replace `c5e964e2-6bb2-2222-1111-3b16ec0e1234` and `78975f9f-2222-1111-1111-29c42ac70000` here.

File: src/azure-cli/azure/cli/command_modules/sql/_params.py
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/sql/_params.py:1400]: boolean or only support true?

File: src/azure-cli/azure/cli/command_modules/sql/_params.py
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/sql/_params.py:1365]: In Azure CLI command guideline, -id is GUID as you can see in https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md#argument-naming-conventions

File: src/azure-cli/azure/cli/command_modules/sql/_help.py
Comment (strehan1993) [reply src/azure-cli/azure/cli/command_modules/sql/_help.py:757]: not real, they are made up

File: src/azure-cli/azure/cli/command_modules/sql/_params.py
Comment (strehan1993) [reply src/azure-cli/azure/cli/command_modules/sql/_params.py:1400]: we support only true in server create, null and false have same behavior. the same has been checked in in powershell as well.

File: src/azure-cli/azure/cli/command_modules/sql/_params.py
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/sql/_params.py:1421]: does it accept any value?

File: src/azure-cli/azure/cli/command_modules/sql/_format.py
Comment (strehan1993) [reply src/azure-cli/azure/cli/command_modules/sql/_format.py:255]: No

File: src/azure-cli/azure/cli/command_modules/sql/_params.py
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/sql/_params.py:1639]: recommend `--enable-auto-rotation`

File: src/azure-cli/azure/cli/command_modules/sql/_params.py
Comment (strehan1993) [reply src/azure-cli/azure/cli/command_modules/sql/_params.py:1421]: it's a boolean, and is used as only an argument -expand-ad-admin

File: src/azure-cli/azure/cli/command_modules/sql/_params.py
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/sql/_params.py:1867]: recommend to use `store_true` if only true is accpected

File: src/azure-cli/azure/cli/command_modules/sql/_params.py
Comment (strehan1993) [reply src/azure-cli/azure/cli/command_modules/sql/_params.py:1639]: i don't own this, we have multiple feature teams working on this. out of scope to make this change

File: src/azure-cli/azure/cli/command_modules/sql/_params.py
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/sql/_params.py:1882]: what about use get_enum_type() for allowed values

File: src/azure-cli/azure/cli/command_modules/sql/_params.py
Comment (strehan1993) [reply src/azure-cli/azure/cli/command_modules/sql/_params.py:1867]: i have tested and recorded tests with this, out of scope to modify now, api accepts true, false and null, so it's a general boolean

File: src/azure-cli/azure/cli/command_modules/sql/_format.py
Comment (strehan1993) [reply src/azure-cli/azure/cli/command_modules/sql/_format.py:255]: no

File: src/azure-cli/azure/cli/command_modules/sql/_params.py
Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/sql/_params.py:1421]: If it is boolean, please use get_three_state_flags()

File: src/azure-cli/azure/cli/command_modules/sql/_params.py
Comment (strehan1993) [reply src/azure-cli/azure/cli/command_modules/sql/_params.py:1882]: I have tested all my scenarios with this, no bandwidth to run server and mi tests again

File: src/azure-cli/azure/cli/command_modules/sql/_params.py
Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/sql/_params.py:1867]: The current definition is not boolean type. It will accept any string.

File: src/azure-cli/azure/cli/command_modules/sql/_params.py
Comment (strehan1993) [reply src/azure-cli/azure/cli/command_modules/sql/_params.py:1365]: I posted with more details on the group chat. Please check all feature owners are there on that chat.

**Review by evelyn-ys — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/sql/_format.py
Comment (strehan1993) [reply src/azure-cli/azure/cli/command_modules/sql/_format.py:255]: They deprecated the boolean so we had to switch to actual values in the version bump

Comment (strehan1993) [reply src/azure-cli/azure/cli/command_modules/sql/_format.py:255]: I posted on the chat group to get answers from the feature owner

File: src/azure-cli/azure/cli/command_modules/sql/_params.py
Comment (strehan1993) [reply src/azure-cli/azure/cli/command_modules/sql/_params.py:1421]: okay sure

Comment (strehan1993) [reply src/azure-cli/azure/cli/command_modules/sql/_params.py:1421]: if this flag is present we expand, we don't support false here

File: src/azure-cli/azure/cli/command_modules/sql/_format.py
Comment (strehan1993) [reply src/azure-cli/azure/cli/command_modules/sql/_format.py:255]: Feature owner is oof till 7/12, we will miss our enterprise promise deadlines if we wait for his answer

File: src/azure-cli/azure/cli/command_modules/sql/_params.py
Comment (strehan1993) [reply src/azure-cli/azure/cli/command_modules/sql/_params.py:1365]: What about User Assigned Identities. They are ids but not typical ARM ids so we can't split them into name/ids. This parameter is called user_assigned_identity_id everywhere i.e. ARM, Powershell and this is an identity resource id

**Review by Juliehzl — APPROVED** [review]
LGTM although there is something to improve, but as there is existing code with similar issue, I approve the PR to unblock the merge.

**Comment by strehan1993** [comment]
> LGTM although there is something to improve, but as there is existing code with similar issue, I approve the PR to unblock the merge.

Ack, we will work on improving the code in the future. @ericshape fyi

**Pull request merged**
